vault/identity_store_oidc_provider.go

```diff
@@ -684,3 +783,130 @@ func (i *IdentityStore) pathOIDCClientExistenceCheck(ctx context.Context, req *l
 
 	return entry != nil, nil
 }
+
+// pathOIDCCreateUpdateProvider is used to create a new named provider or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateProvider(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+
+	var provider provider
+	if req.Operation == logical.UpdateOperation {
+		entry, err := req.Storage.Get(ctx, providerPath+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(&provider); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if issuerRaw, ok := d.GetOk("issuer"); ok {
+		provider.Issuer = issuerRaw.(string)
+	} else if req.Operation == logical.CreateOperation {
+		provider.Issuer = d.GetDefaultOrZero("issuer").(string)
+	}
+
+	if allowedClientIDsRaw, ok := d.GetOk("allowed_client_ids"); ok {
+		provider.AllowedClientIDs = allowedClientIDsRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		provider.AllowedClientIDs = d.GetDefaultOrZero("allowed_client_ids").([]string)
+	}
+
+	if scopesRaw, ok := d.GetOk("scopes"); ok {
+		provider.Scopes = scopesRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		provider.Scopes = d.GetDefaultOrZero("scopes").([]string)
+	}
+
```

Comment:
We'll want to perform some validation on the provided issuer similar to [identity_store_oidc.go#L339-L368](c). Along with that, we should plan to store something similar to the [effectiveIssuer](https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc.go#L417-L422) for each Provider (the path will be different, as we discussed).